core.isph

```diff
@@ -0,0 +1,773 @@
+// line flags.
+const uniform int32 __fast_masked_vload = ISPC_FAST_MASKED_VLOAD_VAL;
+
+const uniform int32 __math_lib = ISPC_MATH_LIB_VAL;
+#endif
+
+typedef uniform int8 *uniform opaque_ptr_t;
+
+#if (ISPC_MASK_BITS == 1)
+#define IntMaskType bool
+#define UIntMaskType bool
+#elif (ISPC_MASK_BITS == 8)
+#define IntMaskType int8
+#define UIntMaskType unsigned int8
+#elif (ISPC_MASK_BITS == 16)
+#define IntMaskType int16
+#define UIntMaskType unsigned int16
+#elif (ISPC_MASK_BITS == 32)
+#define IntMaskType int32
+#define UIntMaskType unsigned int32
+#elif (ISPC_MASK_BITS == 64)
+#define IntMaskType int64
+#define UIntMaskType unsigned int64
+#else
+#error Unknown value of ISPC_MASK_BITS
+#endif
+
+#if (ISPC_MASK_BITS == 1)
+#define UniformMaskType uniform bool
+#elif (ISPC_MASK_BITS == 8)
+#define UniformMaskType uniform int8
+#elif (ISPC_MASK_BITS == 16)
+#define UniformMaskType uniform int16
+#elif (ISPC_MASK_BITS == 32)
+#define UniformMaskType uniform int32
+#elif (ISPC_MASK_BITS == 64)
+#define UniformMaskType uniform int64
+#else
+#error Unknown value of ISPC_MASK_BITS
+#endif
+
+#if (ISPC_POINTER_SIZE == 32)
+#define SizeType int32
+#elif (ISPC_POINTER_SIZE == 64)
+#define SizeType int64
+#else
+#error Unkown ISPC_POINTER_SIZE
+#endif
+
+///////////////////////////////////////////////////////////////////////////
+/* Limits of integral and float types. */
+#ifndef INT8_MAX
+#define INT8_MAX (127)
+#endif
+#ifndef INT16_MAX
+#define INT16_MAX (32767)
+#endif
+#ifndef INT32_MAX
+#define INT32_MAX (2147483647L)
+#endif
+#ifndef INT64_MAX
+#define INT64_MAX (9223372036854775807LL)
+#endif
+#ifndef UINT8_MAX
+#define UINT8_MAX (255U)
+#endif
+#ifndef UINT16_MAX
+#define UINT16_MAX (65535U)
+#endif
+#ifndef UINT32_MAX
+#define UINT32_MAX (4294967295UL)
+#endif
+#ifndef UINT64_MAX
+#define UINT64_MAX (18446744073709551615ULL)
+#endif
+#ifndef INT8_MIN
+#define INT8_MIN (-INT8_MAX - 1)
+#endif
+#ifndef INT16_MIN
+#define INT16_MIN (-INT16_MAX - 1)
+#endif
+#ifndef INT32_MIN
+#define INT32_MIN (-INT32_MAX - 1)
+#endif
+#ifndef INT64_MIN
+#define INT64_MIN (-INT64_MAX - 1)
+#endif
+#ifndef F16_MIN
+#define F16_MIN (6.103515625e-05F16)
+#endif
+#ifndef F16_MAX
+#define F16_MAX (65504.0F16)
+#endif
+#ifndef FLT_MIN
+#define FLT_MIN (1.17549435082228750796873653722224568e-38F)
+#endif
+#ifndef FLT_MAX
+#define FLT_MAX (3.40282346638528859811704183484516925e+38F)
+#endif
+#ifndef DBL_MIN
+#define DBL_MIN (2.22507385850720138309023271733240406e-308D)
+#endif
+#ifndef DBL_MAX
+#define DBL_MAX (1.79769313486231570814527423731704357e+308D)
+#endif
+
+///////////////////////////////////////////////////////////////////////////
+// GEN target specific
+// 4 bytes by default
+#ifndef PREFETCH_DATASIZE_DEFAULT
+#define PREFETCH_DATASIZE_DEFAULT 4
+#endif
+
+// Macros for attributes
+#define NOESCAPE __attribute__((noescape))
+#define ADDRSPACE(N) __attribute__((address_space(N)))
+// #define READONLY __attribute__((memory("read")))
```

Comment:
Is READONLY and READNONE are commented intentionally?

Reply:
As discussed, I went through more or less all function in builtins to check that I am not losing attributes when lifting declarations from *.ll files to *.isph. It also helped to understand which attributes are still to be implemented because of that they are commented.

Reply:
OK, I was a little bit confused. 

- You've supported `noescape` and `address_space` attributes already and they are used in this PR - great.
- You've created issue https://github.com/ispc/ispc/issues/2941 to support "deprecated" but I don't see it anywhere in this PR. Is this attribute essential for the target redesign, or is it just a nice-to-have feature? 
- I don't see an issue created for supporting memory attributes. Please create it.

Reply:
I have created issue for memory attributes https://github.com/ispc/ispc/issues/2956